### PR TITLE
Doc/Adding basic usage example in each model docstring

### LINKED ISSUE #690: Add short examples code snippets for each model in darts documentation

### PR DESCRIPTION
**Pull request by madtoinou**
Fixes #690.

### Summary

Adding short examples in each model docstring, maximizing the covariates usage when supported and pointing out some interesting hyper-parameters when relevant. Let me know if I should extend it to additional methods.

### Other Information

Fixing an inconsistency for `Prophet`; this model supports future covariates but they were not stored during the call to `fit()`, forcing the user to provide them again at inference.

### CODE REVIEW

File: darts/models/forecasting/fft.py [hunk 1]
```diff
@@ -253,6 +253,26 @@ def __init__(
         global trend, and do not perform any frequency filtering:
 
         >>> FFT(required_matches={'month'}, trend='exp')
+
+        Simple usage example, using one of the dataset available in darts
```
Comment (dennisbader) [inline darts/models/forecasting/fft.py:257]: P3

File: darts/models/forecasting/forecasting_model.py
Comment (dennisbader) [inline darts/models/forecasting/forecasting_model.py:1]: these changes shouldn't be in this PR

File: darts/models/forecasting/kalman_forecaster.py
Comment (dennisbader) [inline darts/models/forecasting/kalman_forecaster.py:1]: P1

File: darts/models/forecasting/lgbm.py [hunk 2]
```diff
@@ -114,6 +114,36 @@ def __init__(
             treated as categorical are integer-encoded.
         **kwargs
             Additional keyword arguments passed to `lightgbm.LGBRegressor`.
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/lgbm.py:118]: P3, P4

File: darts/models/forecasting/linear_regression_model.py [hunk 3]
```diff
@@ -96,6 +96,50 @@ def __init__(
             Additional keyword arguments passed to `sklearn.linear_model.LinearRegression` (by default), to
             `sklearn.linear_model.PoissonRegressor` (if `likelihood="poisson"`), or to
             `sklearn.linear_model.QuantileRegressor` (if `likelihood="quantile"`).
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/linear_regression_model.py:100]: P3, P4, P5

File: darts/models/forecasting/nbeats.py [hunk 4]
```diff
@@ -722,6 +722,32 @@ def __init__(
         References
         ----------
         .. [1] https://openreview.net/forum?id=r1ecqn4YwB
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/nbeats.py:726]: P3, P4

File: darts/models/forecasting/nhits.py [hunk 5]
```diff
@@ -659,6 +659,32 @@ def __init__(
         ----------
         .. [1] C. Challu et al. "N-HiTS: Neural Hierarchical Interpolation for Time Series Forecasting",
                https://arxiv.org/abs/2201.12886
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/nhits.py:665]: P3, P4

File: darts/models/forecasting/nlinear.py [hunk 6]
```diff
@@ -351,6 +351,34 @@ def __init__(
         ----------
         .. [1] Zeng, A., Chen, M., Zhang, L., & Xu, Q. (2022).
                Are Transformers Effective for Time Series Forecasting?. arXiv preprint arXiv:2205.13504.
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/nlinear.py:355]: P3, P4

File: darts/models/forecasting/prophet_model.py [hunk 7]
```diff
@@ -124,6 +130,29 @@ def __init__(
             Some optional keyword arguments for Prophet.
             For information about the parameters see:
             `The Prophet source code <https://github.com/facebook/prophet/blob/master/python/prophet/forecaster.py>`_.
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/prophet_model.py:136]: P1, P3
also, prophet uses a daily monthly and yearly seasonality by deafult, maybe not the best example to add this seasonality

File: darts/models/forecasting/random_forest.py [hunk 8]
```diff
@@ -92,6 +92,37 @@ def __init__(
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
         **kwargs
             Additional keyword arguments passed to `sklearn.ensemble.RandomForest`.
+
+        Examples
+        --------
+        Deterministic forecasting, using past/future covariates (optional)
```
Comment (dennisbader) [inline darts/models/forecasting/random_forest.py:98]: Just to be identical to the other examples
```suggestion
```

File: darts/models/forecasting/random_forest.py [hunk 9]
```diff
@@ -92,6 +92,37 @@ def __init__(
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
         **kwargs
             Additional keyword arguments passed to `sklearn.ensemble.RandomForest`.
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/random_forest.py:96]: P3. P4

File: darts/models/forecasting/regression_ensemble_model.py [hunk 10]
```diff
@@ -71,6 +71,40 @@ def __init__(
         References
         ----------
         .. [1] D. H. Wolpert, “Stacked generalization”, Neural Networks, vol. 5, no. 2, pp. 241–259, Jan. 1992
+
+        Examples
+        --------
+        Deterministic forecasting
```
Comment (dennisbader) [inline darts/models/forecasting/regression_ensemble_model.py:77]: P3, P5

also can we use a naive and a linearregression model as forecastingmodels? to make it clear that any type of model can be used

File: darts/models/forecasting/regression_model.py [hunk 11]
```diff
@@ -126,6 +126,24 @@ def __init__(
             Whether the model should use static covariate information in case the input `series` passed to ``fit()``
             contain static covariates. If ``True``, and static covariates are available at fitting time, will enforce
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/regression_model.py:132]: P1

File: darts/models/forecasting/rnn_model.py [hunk 12]
```diff
@@ -383,6 +383,32 @@ def __init__(
         show_warnings
             whether to show warnings raised from PyTorch Lightning. Useful to detect potential issues of
             your forecasting use case. Default: ``False``.
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/rnn_model.py:389]: P3, P4

File: darts/models/forecasting/tcn_model.py [hunk 13]
```diff
@@ -424,6 +424,31 @@ def __init__(
         References
         ----------
         .. [1] https://arxiv.org/abs/1803.01271
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/tcn_model.py:430]: P3. P4

File: darts/models/forecasting/tft_model.py [hunk 14]
```diff
@@ -871,6 +871,33 @@ def __init__(
         ----------
         .. [1] https://arxiv.org/pdf/1912.09363.pdf
         ..[2] Shazeer, Noam, "GLU Variants Improve Transformer", 2020. arVix https://arxiv.org/abs/2002.05202.
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:877]: P3. P4

File: darts/models/forecasting/tft_model.py [hunk 15]
```diff
@@ -871,6 +871,33 @@ def __init__(
...
+        >>> # past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> model = TFTModel(
+        >>>             input_chunk_length=6,
+        >>>             output_chunk_length=6,
+        >>>             n_epochs=5,
+        >>>             )
+        >>> # future_covariates is mandatory for `TFTModel`
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:891]:
```suggestion
        >>> # future_covariates are mandatory for `TFTModel`
```

File: darts/models/forecasting/tft_model.py [hunk 16]
```diff
@@ -871,6 +871,33 @@ def __init__(
         ----------
         .. [1] https://arxiv.org/pdf/1912.09363.pdf
         ..[2] Shazeer, Noam, "GLU Variants Improve Transformer", 2020. arVix https://arxiv.org/abs/2002.05202.
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:873]:
```suggestion
        .. [2] Shazeer, Noam, "GLU Variants Improve Transformer", 2020. arVix https://arxiv.org/abs/2002.05202.
```

File: darts/models/forecasting/tide_model.py [hunk 17]
```diff
@@ -530,6 +530,32 @@ def __init__(
                 http://arxiv.org/abs/2304.08424
         .. [2] T. Kim et al. "Reversible Instance Normalization for Accurate Time-Series Forecasting against
                 Distribution Shift", https://openreview.net/forum?id=cGDAkQo1C0p
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/tide_model.py:536]: P3, P4

File: darts/models/forecasting/tft_model.py [hunk 18]
```diff
@@ -871,6 +871,33 @@ def __init__(
...
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> model = TFTModel(
+        >>>             input_chunk_length=6,
+        >>>             output_chunk_length=6,
+        >>>             n_epochs=5,
+        >>>             )
+        >>> # future_covariates is mandatory for `TFTModel`
+        >>> model.fit(target, past_covariates=past_cov, future_covariates=future_cov)
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:892]: TFTModel is using QuantileRegression by default. We should explain this and take num_samples>>1 and `pred.all_values()`

File: darts/models/forecasting/transformer_model.py [hunk 19]
```diff
@@ -520,11 +520,35 @@ def __init__(
...
         This current implementation is fully functional and can already produce some good predictions. However,
         it is still limited in how it uses the Transformer architecture because the `tgt` input of
-        `torch.nn.Transformer` is not utlized to its full extent. Currently, we simply pass the last value of the
+        `torch.nn.Transformer` is not utilized to its full extent. Currently, we simply pass the last value of the
         `src` input to `tgt`. To get closer to the way the Transformer is usually used in language models, we
         should allow the model to consume its own output as part of the `tgt` argument, such that when predicting
         sequences of values, the input to the `tgt` argument would grow as outputs of the transformer model would be
         added to it. Of course, the training of the model would have to be adapted accordingly.
+
+        Examples
```
Comment (dennisbader) [inline darts/models/forecasting/transformer_model.py:529]: P3, P4

File: darts/models/forecasting/varima.py [hunk 20]
```diff
@@ -71,6 +71,25 @@ def __init__(
                     'transformer': Scaler()
                 }
             ..
+
+        Examples
+        --------
+        >>> from darts.datasets import ETTh2Dataset
```
Comment (dennisbader) [inline darts/models/forecasting/varima.py:77]: P1

File: darts/models/forecasting/xgboost.py [hunk 21]
```diff
@@ -112,6 +112,36 @@ def __init__(
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
         **kwargs
             Additional keyword arguments passed to `xgb.XGBRegressor`.
+
+        Examples
+        --------
+        Deterministic forecasting, using past/future covariates (optional)
```
Comment (dennisbader) [inline darts/models/forecasting/xgboost.py:118]: P3, P4

**Review by dennisbader — CHANGES_REQUESTED** [review]
Very nice, thanks a lot @madtoinou 🚀 
Added a couple of suggestions, mainly to follow a common format

File: darts/models/forecasting/prophet_model.py (hunk 7 above)
Comment (madtoinou) [reply darts/models/forecasting/prophet_model.py:136]: Changed to quarterly, and I added the country holidays in the example.

File: darts/models/forecasting/regression_ensemble_model.py (hunk 10 above)
Comment (madtoinou) [reply darts/models/forecasting/regression_ensemble_model.py:77]: Should I also include covariates in the example or leave it to the ensembling notebook?

File: darts/models/forecasting/tft_model.py (hunk 18 above)
Comment (madtoinou) [reply darts/models/forecasting/tft_model.py:892]: Let me know if the explanations are detailed enough

File: darts/models/forecasting/regression_ensemble_model.py (hunk 10 above)
Comment (dennisbader) [reply darts/models/forecasting/regression_ensemble_model.py]: we can leave it to the notebook

File: darts/models/forecasting/block_rnn_model.py [hunk 22]
```diff
@@ -311,6 +311,30 @@ def encode_year(idx):
...
         .. [1] T. Kim et al. "Reversible Instance Normalization for Accurate Time-Series Forecasting against
                 Distribution Shift", https://openreview.net/forum?id=cGDAkQo1C0p
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
+        >>> from darts.models import BlockRNNModel
+        >>> from darts.utils.timeseries_generation import datetime_attribute_timeseries as dt_attr
+        >>> series = AirPassengersDataset().load()
+        >>> # optionnaly, one-hot encode the months as a past covariates
```
Comment (dennisbader) [inline darts/models/forecasting/block_rnn_model.py:321]: we can use the same as my example for future covariates with `add_length=0`
also, no need to abbreviate `datetime_attribute_timeseries`

File: darts/models/forecasting/block_rnn_model.py [hunk 23]
```diff
@@ -311,6 +311,30 @@ def encode_year(idx):
         ----------
         .. [1] T. Kim et al. "Reversible Instance Normalization for Accurate Time-Series Forecasting against
                 Distribution Shift", https://openreview.net/forum?id=cGDAkQo1C0p
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/block_rnn_model.py:317]: N3 N4

File: darts/models/forecasting/catboost_model.py [hunk 24]
```diff
@@ -97,6 +97,35 @@ def encode_year(idx):
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
         **kwargs
             Additional keyword arguments passed to `catboost.CatBoostRegressor`.
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/catboost_model.py:103]: N3

File: darts/models/forecasting/dlinear.py [hunk 25]
```diff
@@ -414,6 +414,34 @@ def encode_year(idx):
                Are Transformers Effective for Time Series Forecasting?. arXiv preprint arXiv:2205.13504.
         .. [2] T. Kim et al. "Reversible Instance Normalization for Accurate Time-Series Forecasting against
                 Distribution Shift", https://openreview.net/forum?id=cGDAkQo1C0p
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/dlinear.py:420]: N3, N4

File: darts/models/forecasting/fft.py [hunk 26]
```diff
@@ -253,6 +253,26 @@ def __init__(
...
+        >>> from darts.datasets import AirPassengersDataset
+        >>> from darts.models import FFT
+        >>> series = AirPassengersDataset().load()
+        >>> # increase the number of frequency and use a polynomial trend of degree 2
+        >>> model = FFT(
```
Comment (dennisbader) [inline darts/models/forecasting/fft.py:262]: N3

File: darts/models/forecasting/kalman_forecaster.py [hunk 27]
```diff
@@ -75,6 +75,26 @@ def __init__(
                     'transformer': Scaler()
                 }
             ..
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/kalman_forecaster.py:81]: N1, N2

File: darts/models/forecasting/block_rnn_model.py [hunk 28]
```diff
@@ -311,6 +311,30 @@ def encode_year(idx):
...
+        >>> from darts.models import BlockRNNModel
+        >>> from darts.utils.timeseries_generation import datetime_attribute_timeseries as dt_attr
+        >>> series = AirPassengersDataset().load()
```
Comment (dennisbader) [inline darts/models/forecasting/block_rnn_model.py:320]: should we use the weather DS also for BlockRNN as for the other torch models?

File: darts/models/forecasting/nhits.py [hunk 29]
```diff
@@ -671,6 +671,32 @@ def encode_year(idx):
...
+        >>> from darts.models import NHiTSModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # increasing the number of blocks
+        >>> model = NHiTSModel(
```
Comment (dennisbader) [inline darts/models/forecasting/nhits.py:685]: N3

File: darts/models/forecasting/nlinear.py [hunk 30]
```diff
@@ -363,6 +363,34 @@ def encode_year(idx):
...
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # optionally, use future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # predict 6 pressure values using the 12 past values of pressure and rainfall, as well as the 6 temperature
+        >>> # values corresponding to the forecasted period
+        >>> model = NLinearModel(
+        >>>   input_chunk_length=6,
```
Comment (dennisbader) [inline darts/models/forecasting/nlinear.py:381]: N3

File: darts/models/forecasting/prophet_model.py [hunk 31]
```diff
@@ -124,6 +130,33 @@ def __init__(
...
+        >>> from darts.models import Prophet
+        >>> from darts.utils.timeseries_generation import datetime_attribute_timeseries as dt_attr
+        >>> series = AirPassengersDataset().load()
+        >>> # optionally, one-hot encode the months as a future covariates
+        >>> future_cov = dt_attr(series, "month", one_hot=True, add_length=6)
+        >>> # adding a seasonality (yearly and weekly are included by default) and holidays
```
Comment (dennisbader) [inline darts/models/forecasting/prophet_model.py:142]:
```suggestion
        >>> # adding a seasonality (daily, weekly, and yearly are included by default) and holidays
```

File: darts/models/forecasting/prophet_model.py [hunk 32]
```diff
@@ -124,6 +130,33 @@ def __init__(
...
+        >>> model = Prophet(
+        >>>   add_seasonalities={
+        >>>       'name':"quarterly_seasonality",
+        >>>       'seasonal_periods':4,
+        >>>       'fourier_order':5
+        >>>       },
+        >>>   country_holidays="US"
```
Comment (dennisbader) [inline darts/models/forecasting/prophet_model.py:149]:
```suggestion
```

File: darts/models/forecasting/prophet_model.py [hunk 33]
```diff
@@ -124,6 +130,33 @@ def __init__(
             Some optional keyword arguments for Prophet.
             For information about the parameters see:
             `The Prophet source code <https://github.com/facebook/prophet/blob/master/python/prophet/forecaster.py>`_.
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/prophet_model.py:136]: N1 N2 N3

File: darts/models/forecasting/random_forest.py [hunk 34]
```diff
@@ -95,6 +95,36 @@ def encode_year(idx):
...
+        >>> from darts.models import RandomForest
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # optionally, use future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # random forest with 200 trees trained with MAE
+        >>> model = RandomForest(
```
Comment (dennisbader) [inline darts/models/forecasting/random_forest.py:111]: N3

File: darts/models/forecasting/regression_ensemble_model.py [hunk 35]
```diff
@@ -71,6 +71,25 @@ def __init__(
...
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
+        >>> from darts.models import RegressionEnsembleModel, NaiveSeasonal, LinearRegressionModel
+        >>> series = AirPassengersDataset().load()
+        >>> model = RegressionEnsembleModel(
+              forecasting_models = [NaiveSeasonal(K=12), LinearRegressionModel(lags=4)],
+              regression_train_n_points=20
+              )
```
Comment (dennisbader) [inline darts/models/forecasting/regression_ensemble_model.py:83]: incorrect format

File: darts/models/forecasting/regression_ensemble_model.py [hunk 36]
```diff
@@ -71,6 +71,25 @@ def __init__(
         References
         ----------
         .. [1] D. H. Wolpert, “Stacked generalization”, Neural Networks, vol. 5, no. 2, pp. 241–259, Jan. 1992
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
```
Comment (dennisbader) [inline darts/models/forecasting/regression_ensemble_model.py:77]: N3

File: darts/models/forecasting/regression_model.py [hunk 37]
```diff
@@ -129,6 +129,36 @@ def encode_year(idx):
             Whether the model should use static covariate information in case the input `series` passed to ``fit()``
             contain static covariates. If ``True``, and static covariates are available at fitting time, will enforce
             that all target `series` have the same static covariate dimensionality in ``fit()`` and ``predict()``.
+
+        Examples
+        --------
+        >>> from darts.datasets import WeatherDataset
+        >>> from darts.models import RegressionModel
+        >>> from sklearn.linear_model import Ridge
```
Comment (dennisbader) [inline darts/models/forecasting/regression_model.py:137]: N3

File: darts/models/forecasting/rnn_model.py [hunk 38]
```diff
@@ -390,6 +390,32 @@ def encode_year(idx):
...
+        >>> from darts.models import RNNModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # `training_length` > `input_chunk_length` to mimic inference constraints
```
Comment (dennisbader) [inline darts/models/forecasting/rnn_model.py:403]: N3

File: darts/models/forecasting/tcn_model.py [hunk 39]
```diff
@@ -436,6 +436,31 @@ def encode_year(idx):
...
+        >>> from darts.models import TCNModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # `output_chunk_length` must be strictly smaller than `input_chunk_length`
+        >>> model = TCNModel(
```
Comment (dennisbader) [inline darts/models/forecasting/tcn_model.py:450]: N3 N4

File: darts/models/forecasting/tft_model.py [hunk 40]
```diff
@@ -883,6 +883,39 @@ def encode_year(idx):
...
+        >>> from darts.models import TFTModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # by default, TFTModel is trained using a `QuantileRegression`
+        >>> model = TFTModel(
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:899]: N3 N4

File: darts/models/forecasting/tft_model.py [hunk 41]
```diff
@@ -883,6 +883,39 @@ def encode_year(idx):
...
+        >>> from darts.datasets import WeatherDataset
+        >>> from darts.models import TFTModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # by default, TFTModel is trained using a `QuantileRegression`
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:898]:
```suggestion
        >>> # by default, TFTModel is trained using a `QuantileRegression` making it a probabilistic forecasting model
```

File: darts/models/forecasting/tft_model.py [hunk 42]
```diff
@@ -883,6 +883,39 @@ def encode_year(idx):
...
+        >>> model = TFTModel(
+        >>>   input_chunk_length=6,
+        >>>   output_chunk_length=6,
+        >>>   n_epochs=5,
+        >>>   )
+        >>> # future_covariates are mandatory for `TFTModel`
+        >>> model.fit(target, past_covariates=past_cov, future_covariates=future_cov)
+        >>> # using `num_samples > 1`, the model generates a probabilistic forecast
```
Comment (dennisbader) [inline darts/models/forecasting/tft_model.py:906]:
```suggestion
        >>> # TFTModel is by default a probabilistic model; we use `num_samples >> 1` to generate probabilistic forecasts
```

File: darts/models/forecasting/tft_model.py (hunk 18 above)
Comment (dennisbader) [reply darts/models/forecasting/tft_model.py:905]: 👍

File: darts/models/forecasting/tide_model.py [hunk 43]
```diff
@@ -519,6 +519,32 @@ def encode_year(idx):
...
+        >>> from darts.models import TiDEModel
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # optionally, use future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
```
Comment (dennisbader) [inline darts/models/forecasting/tide_model.py:533]: N3

File: darts/models/forecasting/transformer_model.py [hunk 44]
```diff
@@ -532,11 +532,35 @@ def encode_year(idx):
...
+        --------
+        >>> from darts.datasets import WeatherDataset
```
Comment (dennisbader) [inline darts/models/forecasting/transformer_model.py:543]: N3 N4

File: darts/models/forecasting/varima.py [hunk 45]
```diff
@@ -71,6 +71,28 @@ def __init__(
...
+        >>> from darts.models import VARIMA
+        >>> from darts.utils.timeseries_generation import holidays_timeseries
+        >>> # forecasting the High UseFul Load ("HUFL") and Oil Temperature ("OT")
+        >>> series = ETTh2Dataset().load()[:500][["HUFL", "OT"]]
+        >>> # optionally, encode the holidays as a future covariates
```
Comment (dennisbader) [inline darts/models/forecasting/varima.py:82]:
```suggestion
        >>> # optionally, use some future covariates; e.g. encode each timestep whether it is on a holiday
```

File: darts/models/forecasting/varima.py [hunk 46]
```diff
@@ -71,6 +71,28 @@ def __init__(
...
+        >>> future_cov = holidays_timeseries(series.time_index, "CN", add_length=6)
+        >>> # no clear trend in the dataset
+        >>> model = VARIMA(trend="n")
+        >>> model.fit(series)
```
Comment (dennisbader) [inline darts/models/forecasting/varima.py:86]: forgot to use future_covariates

File: darts/models/forecasting/xgboost.py [hunk 47]
```diff
@@ -115,6 +115,36 @@ def encode_year(idx):
...
+        >>> series = WeatherDataset().load()
+        >>> # predicting atmospheric pressure
+        >>> target = series['p (mbar)'][:100]
+        >>> # optionally, use past observed rainfall (pretending to be unknown beyond index 100)
+        >>> past_cov = series['rain (mm)'][:100]
+        >>> # optionally, use future temperatures (pretending this component is a forecast)
+        >>> future_cov = series['T (degC)'][:106]
+        >>> # predict 6 pressure values using the 12 past values of pressure and rainfall, as well as the 6 temperature
+        >>> # values corresponding to the forecasted period
+        >>> model = XGBModel(
```
Comment (dennisbader) [inline darts/models/forecasting/xgboost.py:133]: N3

File: darts/models/forecasting/arima.py
Comment (dennisbader) [inline darts/models/forecasting/arima.py:1]: We have missing models: 



sf_auto_arima.py
sf_auto_ces.py
sf_auto_ets.py
sf_auto_theta.py

File: darts/models/forecasting/tbats_model.py [hunk 48]
```diff
@@ -173,6 +173,23 @@ def __init__(
             See https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
         random_state
             Sets the underlying random seed at model initialization time.
+
+        Examples
+        --------
+        >>> from darts.datasets import AirPassengersDataset
+        >>> from darts.models import TBATS
```
Comment (dennisbader) [inline darts/models/forecasting/tbats_model.py:180]:
```suggestion
        >>> from darts.models import TBATS  # or BATS
```

**Review by dennisbader — CHANGES_REQUESTED** [review]

**Review by dennisbader — APPROVED** [review]
Looks great now, thanks @madtoinou 

I updated the statsforecast model examples and some leftover lambda reference in some models' `add_encoders` descriptions

**Pull request merged**